pkg/inject/sidecar.go

```diff
@@ -251,3 +251,22 @@ func CleanSidecar(deployment *appsv1.Deployment) {
 		}
 	}
 }
+
+// IfNeeded inject the sidecar on deployment only if it is required
+func IfNeeded(instance *appsv1.Deployment, pods *v1.JaegerList) *appsv1.Deployment {
```

Comment:
I don't think this function name is appropriate. I had to check it twice while doing this review, as I would expect it to return an `instance` (not nil): either the original instance, or a changed instance, but not nil. The func documentation also doesn't tell that this might return nil, and what it means.

I think it would be better to change the `IfNeeded` consumers to check `Needed` instead.